packages/plugin-ext/src/main/node/paths/plugin-paths-service.ts

```diff
@@ -14,59 +14,118 @@
  * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
  ********************************************************************************/
 
-import { injectable, inject } from  'inversify';
-import { FileSystem } from '@theia/filesystem/lib/common';
-import { join } from 'path';
+import { injectable, inject } from 'inversify';
+import { FileSystem, FileStat } from '@theia/filesystem/lib/common';
+import * as path from 'path';
+import * as crypto from 'crypto';
 import URI from '@theia/core/lib/common/uri';
 import { isWindows } from '@theia/core';
 import { PluginPaths } from './const';
 import { PluginPathsService } from '../../common/plugin-paths-protocol';
+import { THEIA_EXT, VSCODE_EXT, getTemporaryWorkspaceFileUri } from '@theia/workspace/lib/browser/workspace-service';
 
 // Service to provide configuration paths for plugin api.
 @injectable()
 export class PluginPathsServiceImpl implements PluginPathsService {
 
-    private windowsConfFolders = [PluginPaths.APP_DATA_WINDOWS_FOLDER, PluginPaths.ROAMING_WINDOWS_FOLDER];
-    private linuxConfFolders = [PluginPaths.LINUX_CONF_FOLDER];
+    private windowsDataFolders = [PluginPaths.WINDOWS_APP_DATA_DIR, PluginPaths.WINDOWS_ROAMING_DIR];
```

Comment:
I think we could drop PluginPaths.WINDOWS_APP_DATA_DIR, PluginPaths.WINDOWS_ROAMING_DIR at all. If we want to use .theia, I think we should not use this location pattern.

Reply:
The problem here is that files which starts from `.` aren't hidden in Windows. So we need to add corresponding attribute to the folder or just use AppData (which is already hidden) as Oleksii has suggested.